src/lib/y2network/config.rb

```diff
@@ -83,10 +86,12 @@ def configs
     #
     # @param interfaces [Array<Interface>] List of interfaces
     # @param routing    [Routing] Object with routing configuration
+    # @param dns        [DNS] Object with DNS configuration
```

Comment:
well, nil is not documented. So I suggest to simply use as default value Y2Network::DNS.new and not have nil anywhere.